rocAL/rocAL/source/bounding_box_graph.cpp

```diff
@@ -147,3 +147,159 @@ void BoundingBoxGraph::update_random_bbox_meta_data(MetaDataBatch *input_meta_da
         input_meta_data->get_bb_labels_batch()[i] = bb_labels;
     }
 }
+
+inline void calculate_ious_for_box(float *ious, BoundingBoxCord box, std::vector<float> anchors)
+{
+    BoundingBoxCord _anchor;
+    _anchor.l = anchors[0];
+    _anchor.t = anchors[1];
+    _anchor.r = anchors[2];
+    _anchor.b = anchors[3];
+    ious[0] = ssd_BBoxIntersectionOverUnion(box, _anchor,true);
+    int best_idx = 0;
+    float best_iou = ious[0];
+
+    for (unsigned int anchor_idx = 1; anchor_idx < (anchors.size() / 4); anchor_idx++)
+    {
+        int m = anchor_idx * 4;
+        BoundingBoxCord _anchor;
+        _anchor.l = anchors[m];
+        _anchor.t = anchors[m + 1];
+        _anchor.r = anchors[m + 2];
+        _anchor.b = anchors[m + 3];
+        float x = ssd_BBoxIntersectionOverUnion(box, _anchor, true);
+        ious[anchor_idx] = x;
+        if (ious[anchor_idx] > best_iou)
+        {
+            best_iou = ious[anchor_idx];
+            best_idx = anchor_idx;
+        }
+    }
+    // For best default box matched with current object let iou = 2, to make sure there is a match,
+    // as this object will be the best (highest IoU), for this default box
+    ious[best_idx] = 2.;
+}
+
+inline int find_best_box_for_anchor(unsigned anchor_idx, const std::vector<float> &ious, unsigned num_boxes, unsigned anchors_size)
+{
+    unsigned best_idx = 0;
+    float best_iou = ious[anchor_idx];
+    for (unsigned bbox_idx = 1; bbox_idx < num_boxes; ++bbox_idx)
+    {
+        if (ious[bbox_idx * anchors_size + anchor_idx] >= best_iou)
+        {
+            best_iou = ious[bbox_idx * anchors_size + anchor_idx];
+            best_idx = bbox_idx;
+        }
+    }
+    return best_idx;
+}
+
+void BoundingBoxGraph::update_box_encoder_meta_data(std::vector<float> anchors, pMetaDataBatch full_batch_meta_data, float criteria, bool offset, float scale, std::vector<float> means, std::vector<float> stds)
```

Comment:
It is better to pass anchors as vector of boundingboxes (std::vector<std::tuple<float,float,float,float>> anchors)
If you do that, we can avoidcopying anchors to BoundingBoxCord again where it is used